### DISCUSSION TITLE
How to assert/expect when external app is opened.

### DISCUSSION THREAD

**Discussion by jagadeeshpurplle** [leancodepl/patrol#581] [Q&A] [answered]
Hi patrol community,

I'm using google flutter integration driver to automate the flutter mobile app.

I recently tried patrol to communicate with native elements in flutter app, which looks pretty promising.

I have a use case where clicking on 
- `Mail us` button opens `Gmail app` with subject, body
- `Call us` button opens `Dailer app` with phone number

I want assert/verify that gmail/phone app is opened.
- Is it possible to verify that gmail/phone app package name at that moment? Or any other way to assert this kind of use cases?
- Either verifying the subject, content in gmail compose screen  

Here is the video of app application action: 

https://user-images.githubusercontent.com/25878053/200491878-201db68f-597d-4a52-b04d-888ca8991fe6.mov

@bartekpacia

**Comment by bartekpacia** [comment, marked as answer]
Hi @jagadeeshpurplle 👋

> Is it possible to verify that gmail/phone app package name at that moment? Or any other way to assert this kind of use cases?

No, currently Patrol doesn't have this feature.

I think it might be possible to implement. On iOS, we can easily [check the currently open app](https://stackoverflow.com/questions/73976961/how-to-check-if-any-app-is-installed-during-xctest). I don't know about Android - it would require some research.

> Either verifying the subject, content in gmail compose screen

This is not possible directly as well, but you can try a workaround:

```dart
final titleViews = await $.native.getNativeViews(Selector(textContains: 'Samplee email sender'));
expect(titleViews, hasLength(1));

final contentViews = await $.native.getNativeViews(Selector(textContains: 'Samplee email sender from sender'));
expect(contentViews, hasLength(1));
```
